internal/dlog/calc_test.go

```diff
@@ -76,14 +77,28 @@ func TestCalcZp_BabyStepGiantStep_ElGamal(t *testing.T) {
 }
 
 func TestCalcBN256_BabyStepGiantStep(t *testing.T) {
-	xCheck := big.NewInt(10000000)
-	bound := big.NewInt(10000000)
+
+	bound := big.NewInt(100000000)
+	sampler := sample.NewUniformRange(new(big.Int).Neg(bound), bound)
+
+	xCheck, err := sampler.Sample()
+	if err != nil {
+		t.Fatalf("error when generating random number: %v", err)
+	}
+
 	g1gen := new(bn256.G1).ScalarBaseMult(big.NewInt(1))
 	g2gen := new(bn256.G2).ScalarBaseMult(big.NewInt(1))
 	g := bn256.Pair(g1gen, g2gen)
-	h := new(bn256.GT).ScalarMult(g, xCheck)
+	var h *bn256.GT
```

Comment:
Maybe you can change this to `h := new(bn256.GT)` and in the if/else clauses, do just `h.ScalarMult()`.

Reply:
Good idea. I also just saw that g is generated in a long way and I will just replace it with `g := new(bn256.GT).ScalarBaseMult(big.NewInt(1))`.